From 0dbefc4fa4fc1292fa2c940304a5ec0df204cf98 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Tue, 4 Oct 2022 22:59:16 +0300
Subject: [PATCH] Replace loaL_reg with luaL_Reg according to lua5.1

---
 iwinfo_lua.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/iwinfo_lua.c b/iwinfo_lua.c
index 996f291..39ac2af 100644
--- a/iwinfo_lua.c
+++ b/iwinfo_lua.c
@@ -807,7 +807,7 @@ LUA_WRAP_STRUCT_OP(wext,hardware_id)

 #ifdef USE_WL
 /* Broadcom table */
-static const luaL_reg R_wl[] = {
+static const luaL_Reg R_wl[] = {
 	LUA_REG(wl,channel),
 	LUA_REG(wl,frequency),
 	LUA_REG(wl,frequency_offset),
@@ -840,7 +840,7 @@ static const luaL_reg R_wl[] = {

 #ifdef USE_MADWIFI
 /* Madwifi table */
-static const luaL_reg R_madwifi[] = {
+static const luaL_Reg R_madwifi[] = {
 	LUA_REG(madwifi,channel),
 	LUA_REG(madwifi,frequency),
 	LUA_REG(madwifi,frequency_offset),
@@ -873,7 +873,7 @@ static const luaL_reg R_madwifi[] = {

 #ifdef USE_NL80211
 /* NL80211 table */
-static const luaL_reg R_nl80211[] = {
+static const luaL_Reg R_nl80211[] = {
 	LUA_REG(nl80211,channel),
 	LUA_REG(nl80211,frequency),
 	LUA_REG(nl80211,frequency_offset),
@@ -906,7 +906,7 @@ static const luaL_reg R_nl80211[] = {

 /* Wext table */
 #ifdef USE_WEXT
-static const luaL_reg R_wext[] = {
+static const luaL_Reg R_wext[] = {
 	LUA_REG(wext,channel),
 	LUA_REG(wext,frequency),
 	LUA_REG(wext,frequency_offset),
@@ -938,7 +938,7 @@ static const luaL_reg R_wext[] = {
 #endif

 /* Common */
-static const luaL_reg R_common[] = {
+static const luaL_Reg R_common[] = {
 	{ "type", iwinfo_L_type },
 	{ "__gc", iwinfo_L__gc  },
 	{ NULL, NULL }
