Index: git/package/base-files/files/bin/config_generate
===================================================================
--- git.orig/package/base-files/files/bin/config_generate
+++ git/package/base-files/files/bin/config_generate
@@ -67,9 +67,9 @@ generate_network() {
 
 	[ -n "$ifname" ] || return
 
-	# force bridge for multi-interface devices (and lan)
+	# force bridge for multi-interface devices
 	case "$1:$ifname" in
-		*\ * | lan:*) type="bridge" ;;
+		*\ *) type="bridge" ;;
 	esac
 
 	uci -q batch <<-EOF
