Index: git/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/CMakeLists.txt
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci)
+
+INCLUDE(GNUInstallDirs)
+
+IF(NOT LUAPATH)
+	SET(LUAPATH "${CMAKE_INSTALL_LIBDIR}/lua/5.1")
+ENDIF()
+
+IF(NOT LUA_CFLAGS)
+	FIND_PROGRAM(PKG_CONFIG pkg-config)
+	IF(PKG_CONFIG)
+		EXECUTE_PROCESS(
+			COMMAND pkg-config --silence-errors --cflags lua5.1
+			OUTPUT_VARIABLE LUA_CFLAGS
+			OUTPUT_STRIP_TRAILING_WHITESPACE
+		)
+	ENDIF()
+ENDIF()
+
+ADD_DEFINITIONS(${LUA_CFLAGS})
+
+ADD_SUBDIRECTORY(modules/luci-base)
+ADD_SUBDIRECTORY(modules/luci-mod-admin-full)
+ADD_SUBDIRECTORY(libs/luci-lib-nixio)
+ADD_SUBDIRECTORY(libs/luci-lib-ip)
+ADD_SUBDIRECTORY(libs/luci-lib-jsonc)
+ADD_SUBDIRECTORY(themes/luci-theme-bootstrap)
+ADD_SUBDIRECTORY(applications/luci-app-firewall)
+
Index: git/libs/luci-lib-ip/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/libs/luci-lib-ip/CMakeLists.txt
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-lib-ip C)
+
+FIND_PACKAGE(PkgConfig)
+
+ADD_LIBRARY(ip SHARED src/ip.c)
+SET_TARGET_PROPERTIES(ip PROPERTIES PREFIX "")
+
+TARGET_LINK_LIBRARIES(ip nl-3)
+
+INSTALL(TARGETS ip
+	LIBRARY DESTINATION "${LUAPATH}/luci"
+)
+
Index: git/libs/luci-lib-nixio/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/libs/luci-lib-nixio/CMakeLists.txt
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-lib-nixio C)
+
+ADD_DEFINITIONS(-DNO_TLS)
+
+ADD_LIBRARY(nixio SHARED src/nixio.c src/socket.c src/sockopt.c src/bind.c src/address.c
+	src/protoent.c src/poll.c src/io.c src/file.c src/splice.c src/process.c
+	src/syslog.c src/bit.c src/binary.c src/fs.c src/user.c)
+SET_TARGET_PROPERTIES(nixio PROPERTIES PREFIX "")
+TARGET_LINK_LIBRARIES(nixio crypt ssl)
+
+INSTALL(DIRECTORY root/usr/lib/lua/
+	DESTINATION ${LUAPATH}
+	USE_SOURCE_PERMISSIONS
+)
+
+INSTALL(TARGETS nixio
+	LIBRARY DESTINATION ${LUAPATH}
+)
+
+INSTALL(DIRECTORY root/lib
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/"
+	USE_SOURCE_PERMISSIONS
+)
+
Index: git/modules/luci-base/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/modules/luci-base/CMakeLists.txt
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-base C)
+
+ADD_LIBRARY(parser SHARED src/template_parser.c src/template_utils.c src/template_lmo.c src/template_lualib.c)
+SET_TARGET_PROPERTIES(parser PROPERTIES PREFIX "")
+
+INSTALL(DIRECTORY luasrc/
+	DESTINATION "${LUAPATH}/luci"
+	USE_SOURCE_PERMISSIONS
+)
+
+INSTALL(TARGETS parser
+	LIBRARY DESTINATION "${LUAPATH}/luci/template"
+)
+
+INSTALL(DIRECTORY root/
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/"
+	USE_SOURCE_PERMISSIONS
+)
+
+INSTALL(DIRECTORY htdocs/
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/www"
+	USE_SOURCE_PERMISSIONS
+)
+
Index: git/modules/luci-mod-admin-full/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/modules/luci-mod-admin-full/CMakeLists.txt
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-mod-admin-full C)
+
+ADD_EXECUTABLE(luci-bwc src/luci-bwc.c)
+TARGET_LINK_LIBRARIES(luci-bwc dl)
+
+INSTALL(DIRECTORY luasrc/
+	DESTINATION "${LUAPATH}/luci"
+	USE_SOURCE_PERMISSIONS
+)
+
+INSTALL(TARGETS luci-bwc
+	RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+)
+
+INSTALL(DIRECTORY htdocs/
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/www"
+	USE_SOURCE_PERMISSIONS
+)
+
Index: git/themes/luci-theme-bootstrap/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/themes/luci-theme-bootstrap/CMakeLists.txt
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-theme-bootstrap C)
+
+INSTALL(DIRECTORY luasrc/
+	DESTINATION "${LUAPATH}/luci"
+	USE_SOURCE_PERMISSIONS
+)
+
+INSTALL(DIRECTORY root/
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/"
+	USE_SOURCE_PERMISSIONS
+)
+
+INSTALL(DIRECTORY htdocs/
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/www"
+	USE_SOURCE_PERMISSIONS
+)
+
Index: git/applications/luci-app-firewall/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/applications/luci-app-firewall/CMakeLists.txt
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-app-firewall C)
+
+INSTALL(DIRECTORY luasrc/
+	DESTINATION "${LUAPATH}/luci"
+	USE_SOURCE_PERMISSIONS
+)
Index: git/libs/luci-lib-jsonc/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/libs/luci-lib-jsonc/CMakeLists.txt
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.0)
+
+PROJECT(luci-lib-jsonc C)
+
+FIND_PACKAGE(PkgConfig)
+
+ADD_LIBRARY(jsonc SHARED src/jsonc.c)
+SET_TARGET_PROPERTIES(jsonc PROPERTIES PREFIX "")
+
+TARGET_LINK_LIBRARIES(jsonc json-c)
+
+INSTALL(TARGETS jsonc
+	LIBRARY DESTINATION "${LUAPATH}/luci"
+)
+
