diff --git a/git/openwrt/package/base-files/files/lib/functions/uci-defaults.sh b/git/openwrt/package/base-files/files/lib/functions/uci-defaults.sh
index 7222ff8..03084f5 100755
--- a/git/openwrt/package/base-files/files/lib/functions/uci-defaults.sh
+++ b/git/openwrt/package/base-files/files/lib/functions/uci-defaults.sh
@@ -1,4 +1,4 @@
-#!/bin/ash
+#!/bin/sh
 
 . /lib/functions.sh
 . /usr/share/libubox/jshn.sh
 
